stanley:
  end_position: [166.5, -104.2, 0.0] # 최종 목적지
  stop_initiation_distance: 80 # 최종 목적지에서 이 파라미터만큼 떨어진 위치부터 stop case로 스위칭. stop case부터 최종 목적지까지의 거리에 대한 pid제어가 진행
  switch_stop_initiation_tolerance: 0.5 # # stop case로 스위칭되는 위치로 인식되는 오차범위
  wheel_base: 2.7 # 앞바퀴와 뒷바퀴 사이의 거리(손대면 안됨)
  stanley_gain: 1.0 # stanley control gain(크면 핸들이 확확 꺾임)
  target_velocity: 30 # km/h, 목표 속도
  window_size: 50 # 속도 계획을 짤 때 몇 개의 point를 가지고 곡률을 계산할 것인가
  rate: 50 # Hz, stanley control 주기

pid_control:
  # 속도 오차를 줄이기 위한 pid 제어 (driving case에서만 제어)
  velocity:
    kp: 0.3
    ki: 0.0
    kd: 0.03
  # 위치 오차를 줄이기 위한 pid 제어 (stop case에서만 제어)
  position:
    kp: 0.5
    ki: 0.0
    kd: 0.0

local_path:
  size: 50 # global path의 몇 개의 point를 가지고 local path를 생성할 것인가
  threshold: 10 # 현재 위치와 인덱스의 차이가 threshold보다 큰 포인트는 무시 (교차로에서 혼란 방지)
  rate: 30 # Hz, stanley control 주기

global_path:
  intermediate_point_distance: 1.0 # global path 간격이 클 때 몇 m 간격으로 포인트를 채울 것인가
  rate: 1 # Hz, stanley control 주기
